pkg/client/metrics/metrics.go

```diff
@@ -44,15 +44,35 @@ var (
 			Buckets:   prometheus.DefBuckets,
 		},
 		[]string{"path", "server"},
+
+	)
+
+	// define a counter for API errors for various ErrorTypes
+	apiErrorCount = metrics.NewCounterVec(
+    		&metrics.CounterOpts{
+        		Namespace: "prometheus_adapter",
+        		Subsystem: "prometheus_client",
+        		Name:      "api_errors_total",
+        		Help:      "Total number of API errors",
+    		},
+    		[]string{"error_code", "path", "server"},
```

Comment:
In order to compute an error rate and produce an alert out of this metric, it is usually better to record all the request including the one that succeed.

A common practice is to have a `http_requests_total` metric with a `code` label. In you case I would suggest to have a `api_requests_total` metric that record all the requests and rename the label to `code` and set it to the http status code of the request.

Reply:
ack, makes sense, updated with api_requests_total. thanks for the suggestion.